presto-native-execution/presto_cpp/main/runtime-metrics/RuntimeMetricsReporter.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "presto_cpp/main/runtime-metrics/RuntimeMetricsReporter.h"
+#include "presto_cpp/main/common/Configs.h"
+#include "presto_cpp/main/runtime-metrics/prometheus-reporter/PrometheusReporterUtil.h"
+#include "velox/velox/common/base/StatsReporter.h"
+
+namespace facebook::presto {
+namespace {
+
+std::unordered_map<
+    std::string,
+    std::unique_ptr<facebook::velox::BaseStatsReporter>>&
+metricsReporters() {
+  static std::unordered_map<
+      std::string,
+      std::unique_ptr<facebook::velox::BaseStatsReporter>>
+      metricsReporters;
+  return metricsReporters;
+}
+} // namespace
+
+void registerRuntimeMetricsHttpEndpoint(
+    const std::unique_ptr<http::HttpServer>& httpServer) {
+  // Add compile flags.
+  httpServer->registerGet(
+      "/v1/info/metrics/",
+      [](proxygen::HTTPMessage* /*message*/,
+         const std::vector<std::unique_ptr<folly::IOBuf>>& /*body*/,
+         proxygen::ResponseHandler* downstream) {
+        auto systemConfig = SystemConfig::instance();
+        auto reporterFormat = systemConfig->runtimeMetricsReporterFormat();
+        if (reporterFormat == kPrometheusReporterName) {
```

Comment:
`getSerializedMetrics()` should be a part of `RuntimeMetricsReporter.cpp`
The body here should be independent of reporter Name.